gramps/gui/displaystate.py

```diff
@@ -132,8 +132,9 @@ def push(self, handle):
             if handle in self.mru:
                 self.mru.remove(handle)
             self.mru.append(handle)
+            if make_active:
+                self.index += 1
             self.emit("mru-changed", (self.mru,))
-            self.index += 1
```

Comment:
I've changed the order in which we emit the `mru-changed` signal and increment the index of the active object.
I now need to know the currently active object when updating the MRU menu

Is this likely to cause problems elsewhere?

Reply:
Possibly for Nick's CombinedView and more likely my experimental CardViews which both use a global navigation history object to try to facilitate navigation across different object types in the same view. We had talked about trying to improve the navigation support in Gramps core to better enable and handle things but nothing has been attempted yet and it has been some time since I reviewed what I did.